trade-finance-manager-ui/server/controllers/utilisation-reports/index.ts

```diff
@@ -1,30 +1,122 @@
+const getUtilisationReportsSummaryData = async (userToken: unknown) => {
+  const currentDate = new Date();
+  const oneIndexedMonth = subMonths(currentDate, 1).getMonth() + 1;
+  const year = currentDate.getFullYear();
+
+  const submissionMonth = getIsoMonth(currentDate);
+  const reconciliationSummaryApiResponse = await api.getUtilisationReportsReconciliationSummary(submissionMonth, asString(userToken));
+
+  const reportReconciliationSummary = getReportReconciliationSummaryViewModel(reconciliationSummaryApiResponse);
+  const reportPeriod = getFormattedReportPeriod();
+  const reportDueDate = await getFormattedReportDueDate(userToken);
+
+  return {
+    reportReconciliationSummary,
+    reportPeriod,
+    reportDueDate,
+    month: oneIndexedMonth,
+    year,
+  };
+};
 
 export const getUtilisationReports = async (req: Request, res: Response) => {
   const { userToken, user } = req.session;
 
   try {
-    const submissionMonth = getIsoMonth(new Date());
-    const reconciliationSummaryApiResponse = await api.getUtilisationReportsReconciliationSummary(submissionMonth, asString(userToken));
-
-    const reportReconciliationSummary = getReportReconciliationSummaryViewModel(reconciliationSummaryApiResponse);
-    const reportPeriod = getFormattedReportPeriod();
-    const reportDueDate = await getFormattedReportDueDate(userToken);
+    const { reportReconciliationSummary, reportPeriod, reportDueDate, month, year } = await getUtilisationReportsSummaryData(userToken);
 
     return res.render('utilisation-reports/utilisation-reports.njk', {
       user,
       activePrimaryNavigation: 'utilisation reports',
       reportReconciliationSummary,
       reportPeriod,
       reportDueDate,
+      month,
+      year,
     });
   } catch (error) {
     console.error('Error rendering utilisation reports page', error);
     return res.render('_partials/problem-with-service.njk', { user });
   }
 };
+
+const getReportIdentifiersFromUrl = (body: object, month: number, year: number): ReportIdentifier[] =>
+  Object.keys(body).reduce((reportIdentifiers, key) => {
+    const match = key.match(/(set-status--)(.*)(-)([a-f\d]+)/);
+    if (!match) {
+      return reportIdentifiers;
+    }
+
+    const idType = match.at(2);
+    const id = match.at(4);
+
+    if (!idType || !id) {
+      throw new Error('Failed to parse report identifiers from request body');
+    }
+
+    if (idType === 'reportId') {
+      return [...reportIdentifiers, { id }];
+    }
+
+    if (idType === 'bankId') {
+      return [...reportIdentifiers, { bankId: id, month, year }];
+    }
+
+    throw new Error('Failed to parse report identifiers from request body');
+  }, [] as ReportIdentifier[]);
+
+export type UpdateReportStatusPayload = {
+  reportsWithStatus: ReportWithStatus[];
+  user: TfmSessionUser;
+};
+
+export const updateUtilisationReportStatus = async (req: Request, res: Response) => {
+  const { user, userToken } = req.session;
+  const { status, month, year } = req.query;
```

Comment:
`status` here is set depending on which button is pressed in the client. There is no reason why I can't change this to either `RECONCILIATION_COMPLETED` or `REPORT_NOT_RECEIVED`, but I want to keep it more general as the behaviour when marking as not completed is not consistent

Reply:
I could also change the name to `state` to avoid confusion with `status`

Reply:
I think it makes sense to have a different name so you don't confuse it with the `status` in the API request, as you say. I think would be good to have something that identifies how it's different, whereas `state` is just another way of saying the same thing. Maybe it could be something like `formStatus`?

I think we would also have a constant to document the possible values.

Reply:
I changed this to be called `formButton` as that is where the information comes from. There is also a new constant called `FORM_BUTTON_VALUES` which documents each of the possible expected values